ir/src/constraints/helpers.rs

```diff
@@ -0,0 +1,44 @@
+use super::{list_comprehension::unfold_lc, IdentifierType, SemanticError, SymbolTable};
+use air_script_core::{ConstantType, Expression, ListFoldingValueType, VariableType};
+
+// HELPERS
+// ================================================================================================
+
+/// Builds a list of expressions from a list folding value. The list folding value can be either a
+/// vector, a list comprehension, or an identifier that refers to a vector.
+///
+/// # Errors
+/// Returns an error if:
+/// - the list folding value is an identifier that does not exist in the symbol table
+/// - the list folding value is an identifier that does not refer to a vector
+pub fn build_list_from_list_folding_value(
+    lf_value_type: &ListFoldingValueType,
+    symbol_table: &SymbolTable,
+) -> Result<Vec<Expression>, SemanticError> {
+    match lf_value_type {
```

Comment:
Should we put this in `list_folding.rs` instead? It would follow the structure of the `tests` more closely